### ISSUE TITLE
Exception thrown: 'DiskQueue.Implementation.TruncatedStreamException'

### ISSUE THREAD

**Issue by WhetDawg** [i-e-b/DiskQueue#23]
I've built a DiskQueue.dll using today's code as .NET Standard 2.0 class library.  
I'm calling the library from both a net472 and core6 console app -- same issue occurring in both.
I'm running on Win11 22H2.

Each call to:
`using (IPersistentQueue queue = PersistentQueue.WaitFor(QueuePathName, TimeSpan.FromSeconds(5)))`
results in 
`Exception thrown: 'DiskQueue.Implementation.TruncatedStreamException' in DiskQueue.dll`

![image](https://user-images.githubusercontent.com/5431588/208323177-2ea62596-d4e6-451c-bb84-b78a93ff8bdb.png)

**The code is working but the exception is troubling.  How am I causing a truncated transaction?**

If I manually delete the C:\TEMP\Test queue folder, there's no exception during the first write.

```
        public static void BasicTest()
        {
            bool bWrite = true;
            bool bRead = true;

            string SharedStoreage = @"C:\TEMP";
            string QueueName = "Test";
            string QueuePathName = Path.Combine(SharedStoreage, QueueName);

            PersistentQueue.DefaultSettings.AllowTruncatedEntries = true;   // Default false
            PersistentQueue.DefaultSettings.ParanoidFlushing = true;        // Default true

            //-----------------------------------------------------------------------------------------
            //  Write
            //-----------------------------------------------------------------------------------------
            string msg = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";

            if (bWrite)
            {
                try
                {
                    System.Diagnostics.Debug.WriteLine("--------------------------------------");
                    System.Diagnostics.Debug.WriteLine($"Write Waiting for {QueuePathName}");

                    for (int i = 0; i < 10; i++)    // Add Messages
                    {
                        System.Diagnostics.Debug.WriteLine("");

                        using (IPersistentQueue queue = PersistentQueue.WaitFor(QueuePathName, TimeSpan.FromSeconds(5)))
                        {
                            System.Diagnostics.Debug.WriteLine($"BEFORE - Queue {QueueName} contains #{queue.EstimatedCountOfItemsInQueue} messages.");
                            System.Diagnostics.Debug.WriteLine("Opening Session");

                            using (var session = queue.OpenSession())
                            {
                                System.Diagnostics.Debug.WriteLine($"Write {QueueName}: {msg}");

                                byte[] data = Encoding.ASCII.GetBytes(msg);

                                session.Enqueue(data);

                                session.Flush();
                            }

                            System.Diagnostics.Debug.WriteLine($"AFTER - Queue {QueueName} contains #{queue.EstimatedCountOfItemsInQueue} messages.");
                        }
                    }
                }
                catch (Exception ex)
                {
                    string sMsg = $"Write {QueueName} Exception: {ex}";

                    System.Diagnostics.Debug.WriteLine(sMsg);
                }
            }

```
![image](https://user-images.githubusercontent.com/5431588/208323131-d5d1d212-ba7a-4ba7-b572-8f3b75bdc953.png)

![image](https://user-images.githubusercontent.com/5431588/208323126-70881751-2b8f-4928-bda8-a4dc759c7e5b.png)

I've played thru all 4 permutations of the following settings from an empty queue folder and there's no difference:

            PersistentQueue.DefaultSettings.AllowTruncatedEntries = false;   // Default false
            PersistentQueue.DefaultSettings.ParanoidFlushing = false;        // Default true

**Comment by i-e-b** [comment]
Hi.

TL;DR: These exceptions are not a problem, and it's probably the logging software being overzealous, or needing a configuration change.


This exception is being used for internal signalling, and shouldn't be raised to your application code.

I think you probably have "First chance exceptions" turned on ( https://learn.microsoft.com/en-us/dotnet/api/system.appdomain.firstchanceexception?view=netstandard-2.0 )

Visual Studio has a setting to either ignore or break on these first chance exceptions. Useful blog-post here:
https://devblogs.microsoft.com/devops/understanding-exceptions-while-debugging-with-visual-studio/

I would generally advise having first-chance exceptions turned off unless you need them. I think Microsoft started putting it on by default to help with ASP.Net.

**Comment by i-e-b** [comment]
I have uploaded https://www.nuget.org/packages/DiskQueue/1.6.4
which replaces the exception with some extra logic. This should work as before, but not cause the unexpected error messages.

Please let me know if you have other issues.
